lib/spark.js

```diff
@@ -118,9 +118,17 @@ export default class Spark {
         'Content-Type': 'application/json'
       }
     })
-    await assertOkResponse(res, 'Failed to submit measurement')
-    const { id } = await res.json()
-    console.log('Measurement submitted (id: %s)', id)
+
+    let id
+    try {
+      await assertOkResponse(res, 'Failed to submit measurement')
+      ;({ id } = await res.json())
+      console.log('Measurement submitted (id: %s)', id)
+      Zinnia.activity.info('Measurement submitted')
+    } finally {
+      Zinnia.activity.error('Failed to submit measurement')
+    }
+    
```

Comment:
This does not belong to the `finally` block, but instead to `catch`, right? In your proposal, we print this error every time we submit a measurement.

Also, maybe we can simplify the code a bit - I am not a big fan of `let id` + `;({ id } = `

```js
    try {
      await assertOkResponse(res, 'Failed to submit measurement')
      const { id } = await res.json()
      console.log('Measurement submitted (id: %s)', id)
      Zinnia.activity.info('Measurement submitted')
      return id
    } catch (err) {
      Zinnia.activity.error('Failed to submit measurement')
      throw err
    }
```

(this is subjective though)